py/dml/dmlparse.py

```diff
@@ -312,9 +312,9 @@ def array_list(t):
 
 @prod
 def object_regarray(t):
-    'object : REGISTER objident array_list sizespec offsetspec maybe_istemplate object_spec'
-    t[0] = ast.object_(site(t), t[2], 'register', t[3],
-                       t[4] + t[5] + t[6] + t[7])
+    'object : maybe_extension REGISTER objident array_list sizespec offsetspec maybe_istemplate object_spec'
```

Comment:
do we really want to allow `in register x size 4 @ 4711 { }` ? all the special syntax is a clear declaration marker to me.

I suggest `in KIND NAME [INDICES] { ... }` as the only allowed form, analogous to the grammar for `in each xyz { ... }`

BTW: should we forbid the `[_ < ...]` index forms outside extensions when the provisional is active?